File: storage/rdb/src/main/java/project/dailyge/entity/anniversary/AnniversaryJpaEntity.java
package project.dailyge.entity.anniversary;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import project.dailyge.entity.BaseEntity;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@Entity(name = "anniversaries")
public class AnniversaryJpaEntity extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "date")
    private LocalDateTime date;

    @Column(name = "remind")
    private boolean remind;

    @Column(name = "emoji_id")
    private Long emojiId;

    @Column(name = "user_id")
    private Long userId;

    public AnniversaryJpaEntity(
        final String name,
        final LocalDateTime date,
        final boolean remind,
        final Long emojiId,
        final Long userId
    ) {
        this(null, name, date, remind, emojiId, userId);
    }

    public AnniversaryJpaEntity(
        final Long id,
        final String name,
        final LocalDateTime date,
        final boolean remind,
        final Long emojiId,
        final Long userId
    ) {
        this.id = id;
        this.name = name;
        this.date = date;
        this.remind = remind;
        this.emojiId = emojiId;
        this.userId = userId;
    }

    public void update(
        final String name,
        final LocalDateTime date,
        final boolean remind,
        final Long emojiId
    ) {
        this.name = name;
        this.date = date;
        this.remind = remind;
        this.emojiId = emojiId;
    }

    public void delete() {
        this.deleted = true;
    }
}

File: storage/rdb/src/main/kotlin/project/dailyge/entity/anniversary/AnniversaryJpaEntity.kt
package project.dailyge.entity.anniversary

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import project.dailyge.entity.BaseEntity
import java.time.LocalDateTime

@Entity(name = "anniversaries")
class AnniversaryJpaEntity(
    @Column(name = "name")
    private var _name: String,

    @Column(name = "date")
    private var _date: LocalDateTime,

    @Column(name = "remind")
    private var _remind: Boolean? = false,

    @Column(name = "emoji_id")
    private var _emojiId: Long? = null,

    @Column(name = "user_id")
    val userId: Long,
) : BaseEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    val name: String
        get() = _name

    val date: LocalDateTime
        get() = _date

    val dateAsString: String
        get() = _date.toLocalDate().toString()

    val remind: Boolean
        get() = _remind ?: false

    val emojiId: Long?
        get() = _emojiId

    fun update(
        name: String,
        date: LocalDateTime,
        remind: Boolean,
        emojiId: Long? = null,
    ) {
        this._name = name
        this._date = date
        this._remind = remind
        this._emojiId = emojiId
    }

    fun delete() {
        updateDeletedStatus()
    }
}
